vispy/visuals/volume.py

```diff
@@ -416,23 +416,55 @@ class VolumeVisual(Visual):
         For instance: if the texture data was last scaled from 0-1, and the clims are set to
         0.4-0.5, then a texture rescale will be triggered if ``clim_range_threshold < 0.1``.
         To prevent rescaling, set this value to 0.  To *always* rescale, set the value to
-        >= 1.  By default, 0.2
+        >= 1.  By default, 0.2 This has no effect if ``texture_format`` is not ``None``.
     emulate_texture : bool
         Use 2D textures to emulate a 3D texture. OpenGL ES 2.0 compatible,
         but has lower performance on desktop platforms.
     interpolation : {'linear', 'nearest'}
-        Selects method of image interpolation. 
+        Selects method of image interpolation.
+    texture_format: numpy.dtype | str | None
+        How to store data on the GPU. OpenGL allows for many different storage
+        formats and schemes for the low-level texture data stored in the GPU.
+        Most common is unsigned integers or floating point numbers.
+        Unsigned integers are the most widely supported while other formats
+        may not be supported on older versions of OpenGL, WebGL
+        (without enabling some extensions), or with older GPUs.
+        Default value is ``None`` which means data will be scaled on the
+        CPU and the result stored in the GPU as an unsigned integer. If a
+        numpy dtype object, an internal texture format will be chosen to
+        support that dtype and data will *not* be scaled on the CPU. Not all
+        dtypes are supported. If a string, then
+        it must be one of the OpenGL internalformat strings described in the
+        table on this page: https://www.khronos.org/registry/OpenGL-Refpages/gl4/html/glTexImage2D.xhtml
+        The name should have `GL_` removed and be lowercase (ex.
+        `GL_R32F` becomes ``'r32f'``). Lastly, this can also be the string
+        ``'auto'`` which will use the data type of the provided volume data
+        to determine the internalformat of the texture.
+        When this is specified (not ``None``) data is scaled on the
+        GPU which allows for faster color limit changes. Additionally, when
+        32-bit float data is provided it won't be copied before being
+        transferred to the GPU. Note this visual is limited to "luminance"
+        formatted data (single band). This is equivalent to `GL_RED` format
+        in OpenGL 4.0.
     """
 
     _interpolation_names = ['linear', 'nearest']
-
-    def __init__(self, vol, clim=None, method='mip', threshold=None, 
+    _texture_dtype_format = {
+        np.float32: 'r32f',
+        np.float64: 'r32f',
```

Comment:
This will induce precision lost, does this need a warning or exception  ? Is it allowed ?

Reply:
Good point. Warning would be good.

Reply:
This now uses the same check function as the ImageVisual. Thanks for pointing out the issue.